recipes/lz4/all/conanfile.py

```diff
@@ -79,17 +80,48 @@ def build(self):
         cmake.configure(build_script_folder=self._cmakelists_folder)
         cmake.build()
 
+    def _create_cmake_module_alias_targets(self, module_file, targets):
+        content = ""
+        for alias, aliased in targets.items():
+            content += textwrap.dedent("""\
+                if(TARGET {aliased} AND NOT TARGET {alias})
+                    add_library({alias} INTERFACE IMPORTED)
+                    set_property(TARGET {alias} PROPERTY INTERFACE_LINK_LIBRARIES {aliased})
+                endif()
+            """.format(alias=alias, aliased=aliased))
+        save(self, module_file, content)
+
+    @property
+    def _module_file_rel_path(self):
+        return os.path.join("lib", "cmake", f"conan-official-{self.name}-targets.cmake")
+
     def package(self):
         copy(self, "LICENSE", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
         cmake = CMake(self)
         cmake.install()
+        if Version(self.version) >= "1.9.4":
+            rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
         rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
         rmdir(self, os.path.join(self.package_folder, "share"))
 
+        if Version(self.version) >= "1.9.4":
+            # TODO: to remove in conan v2 once legacy generators removed
+            self._create_cmake_module_alias_targets(
+                os.path.join(self.package_folder, self._module_file_rel_path),
+                {"lz4": "LZ4::lz4" if self.options.shared else "LZ4::lz4_static"}
```

Comment:
??

Reply:
Yes, I copied it from other recipes but have yet to learn what the purpose is and when to use this. Is there documentation on functions like these somewhere?

What would be the proper fix here? Completely removing the function call because it is not needed?

Reply:
It's needed, I've edited my review with a code suggestion.

Reply:
well too late, it has been merged, I'll open a PR

Reply:
https://github.com/conan-io/conan-center-index/pull/12953